### DISCUSSION TITLE
怎么能避免重复增加Assembly

### DISCUSSION THREAD

**Discussion by TimonYang** [dotnetcore/Natasha#216] [Q&A] [answered]
新使用Natasha，总体非常好用，但是现在有一个小问题，我需要时不时添加新的类型，目前是这样做的
            
            //Assembly[] assemblies = AppDomain.CurrentDomain.GetAssemblies();
            AssemblyCSharpBuilder oop = new AssemblyCSharpBuilder("MyDynamicAssembly");
            oop.Domain = NatashaManagement.CreateRandomDomain();

            oop.Add("new type info");
            Assembly assembly = oop.GetAssembly();
            var func = NDelegate.UseDomain(oop.Domain).Func<object>("Code");
            var result = func();

            func.DisposeDomain();`
但这样做的话，每次调用后，都会新增一个MyDynamicAssembly，所以我的问题是有没有办法去掉之前旧的MyDynamicAssembly，或者有没有其他的方法向之前的MyDynamicAssembly添加新的类？
求救![E7F07EC0](https://github.com/dotnetcore/Natasha/assets/28781772/14b91383-c089-47de-a032-911eaf57f287)

**Comment by NMSAzulX** [comment, marked as answer]
1. 目前来说不能合并程序集，所以无法向程序集中追加内容。运行时不支持。
2. 如果需要经常动态生成同样的类，需要缓存起来， 比如把生成的委托/类型/程序集放在静态区或者字典中。
3. 默认是可以访问同一域内所有的程序集。 只要在同一个域内，你的代码引用动态生成的类是可以的。

**Reply by TimonYang** [reply]
了解，感谢指导。

**Comment by NMSAzulX** [comment]
之前生成的动态程序集只能通过 dispose 方法卸载，如果有强引用是卸载不掉的。 assmebly/type/func.dispose
